main.py

```diff
@@ -151,12 +155,20 @@ def generate_predictions(methods: list[str], datasets: list[str]) -> Generator[R
                 predictions_per_dataset.append(report.prediction)
                 reports_per_dataset.append(report)
 
+                #to store R2
+                prediction = __predictors[method_name](dataset)
+                metrics = calculate_metrics(prediction)['r_squared']
+                metrics_stat.append(metrics)
+
+
             if len(predictions_per_dataset) > 0:
                 comparison_plot_multi(dataset.values.loc[training_index, :], predictions_per_dataset)
 
-            results_store.append(reports_per_dataset)
-            yield reports_per_dataset
-        
+            #appends only the cities where R2 > 10
+            if (np.array(metrics_stat)>-10).all():
+                results_store.append(reports_per_dataset)
+                yield reports_per_dataset
```

Comment:
`yield  reports_per_dataset` doesn't need to be indented- it's only the results_store we want to block.

Reply:
Right, it got accidentally in the middle. I have just changed it.